File: MyApplication2/app/src/main/java/com/example/myapplication/MainActivity2.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Toast;

public class MainActivity2 extends AppCompatActivity {
    private EditText nombres, edad;
    RadioGroup rg;
    RadioButton rb;
    private Button button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);
        this.nombres = findViewById(R.id.editTextTextPersonName);
        this.rg =  findViewById(R.id.radiogroup);
        this.edad =  findViewById(R.id.editTextTextPersonName2);
        this.button =  findViewById(R.id.button);
        int id=rg.getCheckedRadioButtonId();
        button.setOnClickListener(view -> {
            rb =  findViewById(id);
            Intent i = new Intent(MainActivity2.this, MainActivity3.class);
            i.putExtra("Nombre",nombres.getText().toString());
            i.putExtra("Edad",edad.getText().toString());
            i.putExtra("Genero",rb.getText().toString());
            startActivity(i);
        });
    }
}

File: SharedPreferencesAristiDevsExample/app/src/main/java/com/example/sharedpreferencesaristidevsexample/MainActivity2.kt
package com.example.sharedpreferencesaristidevsexample

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.CheckBox
import android.widget.EditText
import android.widget.TextView
import com.example.sharedpreferencesaristidevsexample.UserVipApplication.Companion.prefss

class MainActivity2 : AppCompatActivity() {
    private lateinit var tvName2: TextView
    private lateinit var btnContinue: Button
    private lateinit var etName2: EditText
    private lateinit var cbVip: CheckBox

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        btnContinue = findViewById(R.id.btnContinue)
        etName2 = findViewById(R.id.etName2)
        tvName2 = findViewById(R.id.tvName2)
        cbVip = findViewById(R.id.cbVip)
        initUI()
        checkUserValues()
    }

    private fun checkUserValues() {
        if (prefss.getName().isNotEmpty()) {
            goToDetail()
        }
    }

    private fun initUI() {
        btnContinue.setOnClickListener { accessToDetail() }
    }

    private fun accessToDetail() {
        if (etName2.text.toString().isNotEmpty()) {
            // Guardar usuario
            prefss.saveName(etName2.text.toString())
            prefss.saveVIP(cbVip.isChecked)
            goToDetail()
        } else {
        }
    }

    fun goToDetail() {
        startActivity(Intent(this, ResultActivity::class.java))
    }
}